src/tcgwars/logic/impl/gen7/CosmicEclipse.groovy

```diff
@@ -873,9 +875,9 @@ public enum CosmicEclipse implements LogicCardInfo {
             text "The attacks of your Pokémon-GX in play that evolve from Eevee do 30 more damage to your opponent's Active Pokémon (before applying Weakness and Resistance). You can't apply more than 1 Power Cheer Ability at a time."
             delayedA {
               after PROCESS_ATTACK_EFFECTS, {
-                if (bg.em().retrieveObject("Power_Cheer") != bg.turnCount){
+                if (bg.em().retrieveObject("Power_Cheer") != bg.turnCount && ef.attacker.owner == self.owner && ef.attacker.pokemonGX && ef.attacker.realEvolution ){
```

Comment:
Mostly left it as ef.attacker out of keeping everything consistent in that level (prior to the `bg.em().each`).

But not 100% sure it would work, as iirc `it` for this idea is part of the `bg.em().each` block (and wouldn't be accesible here yet).